discord/interactions.py

```diff
@@ -261,20 +261,32 @@ def _from_data(self, data: InteractionPayload):
             except KeyError:
                 pass
 
-        if channel := data.get("channel"):
-            if (ch_type := channel.get("type")) is not None:
-                factory, ch_type = _threaded_channel_factory(ch_type)
+        channel = data.get("channel")
+        data_ch_type: int | None = None
+        if channel:
+            data_ch_type = channel.get("type")
```

Comment:
This seemed better to me:
```suggestion
        data_ch_type: int | None = channel.get("type") if channel else None
```